common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -143,13 +143,19 @@ public BrokerRequest brokerRequestFromSilentOperationParameters(@NonNull final A
     }
 
     @NonNull
-    private static AbstractAuthenticationScheme getAuthenticationScheme(@NonNull final BrokerRequest request) {
+    private static AbstractAuthenticationScheme getAuthenticationScheme(
+            @NonNull final Context context,
+            @NonNull final BrokerRequest request) {
         final AbstractAuthenticationScheme requestScheme = request.getAuthenticationScheme();
 
         if (null == requestScheme) {
             // Default assumes the scheme is Bearer
             return new BearerAuthenticationSchemeInternal();
         } else {
+            if (requestScheme instanceof PopAuthenticationSchemeInternal) {
+                ((PopAuthenticationSchemeInternal) requestScheme).setContext(context);
```

Comment:
Add a comment explaining what's going on here.

It would be preferable to have some kind of `SkewProvider` object here that removes the tight-coupling between the pop auth scheme and the data source

Reply:
Tidied up a bit in: dff13326e369b599f0ea4aa483c401431b56076b